src/Microsoft.ML.Data/DataView/TypedCursor.cs

```diff
@@ -217,6 +217,90 @@ public RowCursor<TRow>[] GetCursorSet(Func<int, bool> additionalColumnsPredicate
                  .ToArray();
         }
 
+        public static void ValidateMemberInfo(MemberInfo memberInfo, IDataView data)
```

Comment:
Can the methods that you introduced here in `TypedCursor.cs` be `private`? If there's no reason to have them public, I guess it's better to maintain them as private.